align_anything/evaluation/benchmarks/MMLU/vllm_eval.py

```diff
@@ -0,0 +1,243 @@
+                eval_case = {
+                    'question': correct_answer['prompt'],
+                    'choices': correct_answer['choices'],
+                    'correct_answer': correct_answer['answer'],
+                    'answer_logprobs': response['answer_logprobs'],
+                    'chosen_answer': chosen_answer
+                }
+                if judge_answer(correct_answer['answer'], chosen_answer, response['answer']):
+                    cnt_match += 1
+                    eval_case['result'] = True
+                    true_cases.append(eval_case)
+                else:
+                    eval_case['result'] = False
+                    false_cases.append(eval_case)
+                break
+        if flag_fail:
+            cnt_fail += 1
+        else:
+            flag_fail = True
+        
+    return cnt_match, cnt_sum, true_cases, false_cases
+
+def get_chosen_answer(logprobs: List[Dict[str, Any]], candidate_answers: List[str]):
+    answer_logprobs = {}
+    for logprob in logprobs:
+        key = next(iter(logprob.values())).decoded_token
+        value = next(iter(logprob.values())).logprob
+        if key in candidate_answers:
+            answer_logprobs[key] = value
+    for label in candidate_answers:
+        if label not in answer_logprobs.keys():
+            answer_logprobs[label] = float('-inf')
+    return answer_logprobs
+
+def judge_answer(correct_answer, chosen_answer, response):
+    if correct_answer == chosen_answer:
+        return True
+    match = re.search(r'(?<![a-zA-Z])[A-Z](?![a-zA-Z])', response)
+    if match:
+        return correct_answer == match.group()
+    return False
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    _, unparsed_args = parser.parse_known_args()
+    keys = [k[2:] for k in unparsed_args[0::2]]
+    values = list(unparsed_args[1::2])
+    unparsed_args = dict(zip(keys, values))
+    logger = EvalLogger('Evaluation')
+
+    dict_configs, infer_configs = read_eval_cfgs('mmlu', 'vLLM')
+
+    try:
+        assert dict_configs or infer_configs, "Config file does not exist or is incomplete."
+    except AssertionError as e:
+        logger.log('error', "Config file is not exist or incomplete.")
+        exit()
+    
+    for k, v in unparsed_args.items():
+        dict_configs = update_dict(dict_configs, custom_cfgs_to_dict(k, v))
+        infer_configs = update_dict(infer_configs, custom_cfgs_to_dict(k, v))
+    
+    dict_configs, infer_configs = dict_to_namedtuple(dict_configs), dict_to_namedtuple(infer_configs)
+    model_config = dict_configs.default.model_cfgs
+    eval_configs = dict_configs.default.eval_cfgs
+    logger.log_dir = eval_configs.output_dir
+    dataloader = MMLUDataLoader(dict_configs)
+    test_data = dataloader.load_dataset()
+    eval_module = MMLUGeneratorVLLM(model_config, infer_configs)
+    raw_outputs = eval_module.eval(test_data, eval_configs)
```

Comment:
Please check the code. The speed for evaluating MMLU using vLLM is slow. See #35

Reply:
Thanks for your question! I have replied it in #35